src/channel/socketio-channel.ts

```diff
@@ -107,35 +110,47 @@ export class SocketIoChannel extends Channel {
     /**
      * Bind the channel's socket to an event and store the callback.
      */
-    on(event: string, callback: Function): void {
-        let listener = (channel, data) => {
-            if (this.name == channel) {
-                callback(data);
-            }
-        };
+    on(event: string, callback: Function): SocketIoChannel {
```

Comment:
I changed the return type of this method to be analogous to `PusherChannel.on()`.